pkg/cli/prompt/text/text_test.go

```diff
@@ -137,11 +137,7 @@ func Test_E2E(t *testing.T) {
 		tm.Type("abcd")
 		tm.Send(tea.KeyMsg{Type: tea.KeyEnter})
 		tm.WaitFinished(t, teatest.WithFinalTimeout(waitTimeout))
-		bts, err := io.ReadAll(tm.FinalOutput(t))
-		require.NoError(t, err)
 
-		output = normalizeOutput(bts)
-		require.Empty(t, strings.TrimSpace(output)) // Output sometimes contains a single space.
```

Comment:
BTW: what this is doing is verifying that the output is blank after the user confirms the prompt or cancels. This is intentional because we want to hide the display after the user is done.

However, if this is the thing that's causing the test to fail I don't think it's critical to cover in a test. We've seen lots of issues using the test framework (teatest) - so over time these tests have been simplified to try and keep them reliable. 

---

I tried your fix with:

```
go test -count=10000 -run=Test_E2E -failfast ./pkg/cli/prompt/text
```

and it worked for me.

Reply:
Can you also update [here](https://github.com/radius-project/radius/pull/7722/files#diff-a95b3c117c873821d9473cd582ee12f2ffccb36c71d5fd41b1a3897f0268d957L163-L167) and [here](https://github.com/radius-project/radius/pull/7722/files#diff-a95b3c117c873821d9473cd582ee12f2ffccb36c71d5fd41b1a3897f0268d957L116-L120) where we have the same test logic?

If it's wrong for this testcase, then it's probably wrong for the others.

Reply:
> Can you also update [here](https://github.com/radius-project/radius/pull/7722/files#diff-a95b3c117c873821d9473cd582ee12f2ffccb36c71d5fd41b1a3897f0268d957L163-L167) and [here](https://github.com/radius-project/radius/pull/7722/files#diff-a95b3c117c873821d9473cd582ee12f2ffccb36c71d5fd41b1a3897f0268d957L116-L120) where we have the same test logic?
> 
> If it's wrong for this testcase, then it's probably wrong for the others.

Thank you for looking at this PR! 🥇 🚀 

I pushed changes to modify the other two tests. I also refactored to reduce some duplication, but I'm still learning this code so if you disagree with the style I'm happy to change it back to the previous test style.

> BTW: what this is doing is verifying that the output is blank after the user confirms the prompt or cancels. This is intentional because we want to hide the display after the user is done.

That makes a lot of sense for what to test. In the "confirm value" test case where we send "abcd" to the program, if we sleep long enough, the result of `TestModel.FinalOutput()` is this: `"test prompt\n> abcd            \n "`. 

> However, if this is the thing that's causing the test to fail I don't think it's critical to cover in a test. We've seen lots of issues using the test framework (teatest) - so over time these tests have been simplified to try and keep them reliable.

The test framework may need an additional feature to enable these tests. The behavior I observed is that when sending text to the test model, the `TestModel.out` reader is not synchronously updated with program output, I think because it is attached to the `stdout` in the program, which happens in the setup that is performed by `teatest`. (teatest.go, line 131.) The reader is not updated synchronously, which makes logical sense, but there ought to be a way to wait for the result. 
Running `teatest.WithFinalTimeout()`, `TestModel.FinalOutput()`, or `TestModel.FinalModel()` does not force the test to wait until the `stdout` is finished writing to the stream, which means that calling `TestModel.FinalOutput()` is not deterministic.

Reply:
> The test framework may need an additional feature to enable these tests. The behavior I observed is that when sending text to the test model, the TestModel.out reader is not synchronously updated with program output, I think because it is attached to the stdout in the program, which happens in the setup that is performed by teatest. (teatest.go, line 131.) The reader is not updated synchronously, which makes logical sense, but there ought to be a way to wait for the result.

Thanks for digging into. I wondered if this was that was the case - specifically that `FinalOutput` is flaky. We've had a bit of trouble with these tests/this-framework. I'm glad that this code is testable at all, so it seems OK to work around some of the limitations!